src/block/tile/Sign.php

```diff
@@ -111,5 +151,8 @@ public function setEditorEntityRuntimeId(?int $editorEntityRuntimeId) : void{
 
 	protected function addAdditionalSpawnData(CompoundTag $nbt) : void{
 		$nbt->setString(self::TAG_TEXT_BLOB, implode("\n", $this->text->getLines()));
+		$nbt->setString(self::TAG_TEXT_COLOR, $this->textColor->name());
+		$nbt->setByte(self::TAG_GLOWING_TEXT, $this->glowingText ? 1 : 0);
+		$nbt->setByte("TextIgnoreLegacyBugResolved", 1); //workaround for glowing text
```

Comment:
Why is this necessary?

Reply:
Without this, the Sign text won't be glowing.

Reply:
I don't think we can just toss this in and forget about it. If users load vanilla worlds, whatever this is working around should be accommodated on the PM side as well. I believe this has something to do with black glowing text.

Reply:
https://bugs.mojang.com/browse/MCPE-117835